codex/stores/fsstore.nim

```diff
@@ -43,24 +43,31 @@ method getBlock*(
   ## Get a block from the stores
   ##
 
+  trace "Getting block from filestore", cid
   if cid.isEmpty:
     trace "Empty block, ignoring"
     return cid.emptyBlock.success
 
-  if cid in self.cache:
-    return await self.cache.getBlock(cid)
-
-  if cid notin self:
-    return Block.failure("Couldn't find block in fs store")
+  # Try to get this block from the cache
+  let cachedBlock = await self.cache.getBlock(cid)
+  if cachedBlock.isOK:  # TODO: check for success and non-emptiness
+    return cachedBlock
 
+  # Read file contents
   var data: seq[byte]
-  let path = self.blockPath(cid)
-  if (
-    let res = io2.readFile(path, data);
-    res.isErr):
-    let error = io2.ioErrorMsg(res.error)
-    trace "Cannot read file from fs store", path , error
-    return Block.failure("Cannot read file from fs store")
+  let
+    path = self.blockPath(cid)
+    res = io2.readFile(path, data)
+
+  # TODO: If file doesn't exist - return empty block,
```

Comment:
We can't return empty block here, because empty block is not the same as block doesn't exist. An empty block has special semantics in the system.

Reply:
Note that the `getBlock` return type will be `Future[?! (?Block)]`, and by empty block I meant here `Block.none : ?Block`. It will be implemented in next commits in this PR

Reply:
Ah, I see, that makes sense 👍